iothub/device/src/Transport/RetryDelegatingHandler.cs

```diff
@@ -458,7 +474,7 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
         /// <summary>
         /// Implicit open handler.
         /// </summary>
-        private async Task EnsureOpenedAsync(CancellationToken cancellationToken)
+        private async Task EnsureOpenedAsync(CancellationToken cancellationToken, TimeSpan? timeout = null)
```

Comment:
Shouldn't have both `CancellationToken` and `TimeSpan`.  If required `CancellationToken` should be created from `TimeSpan`.